src/initilization-context.ts

```diff
@@ -18,8 +18,12 @@ export class InitializationContext {
 		this.tasks.add(task);
 		task.then(() => {
 			this.tasks.delete(task);
-			if (this.tasks.size === 0)
-				this.waiting.forEach(done => done());
+			// allow additional tasks to be queued as a result of this one
+			Promise.resolve().then(() => {
+				if (this.tasks.size === 0)
+					while (this.waiting.length > 0)
+						this.waiting.shift()();
+			});
 		});
 	}
 
```

Comment:
Should the `waiting.push` below not happen if the task size is 0? Not sure if it matters, but it at least looks like a lurking bug.

Reply:
Fixed